### PR TITLE
Links and meta improvements

### PR DESCRIPTION
**Pull request by pangratz**
[Rendered](https://github.com/pangratz/rfcs/blob/ember-data-links-meta-improvements/text/0000-ember-data-links-meta-improvements.md)

cc @tchak @wecc 

---

_This RFC was born in Greece during the [Ember Thalassa](http://blog.bookingsync.com/en/news/2016/5/20/bookingsync-proud-organizer-of-ember-thalassa-greece) event, sponsored by [bookingsync.com](https://www.bookingsync.com)._

### CODE REVIEW

**Comment by pangratz** [comment]
I started working on a proof of concept https://github.com/pangratz/ember-data-meta-links-improvements.

**Comment by pangratz** [comment]
I gave this more thought and after realizing that JSON-API allows different locations for meta on a JSON-API document (especially relevant for a single resource), I've updated this RFC:

- replace `RecordArrayReference` in favor of `DocumentReference`
- outline different locations of meta
- add code sample for different meta of a resource
- add unresolved question for additional `extractMeta` hook on `RESTSerializer`

File: text/0000-ember-data-links-meta-improvements.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+```js
+// app/models/books.js
+import Model from "ember-data/model";
+
+const Book = Model.extend({});
+
+Book.reopenClass({
+
+  didReceiveData(recordRef) {
```
Comment (jvanantwerp) [inline text/0000-ember-data-links-meta-improvements.md]: FWIW, this is very similar to the way we are handling this case in our hacked-up version of Ember Data we are currently using to address this problems.  +1 to the approach from our POV.

**Comment by dbbk** [comment]
Is there anything we can do to move this along? I can't use Ember Data until there's link-based relationship saving :(

**Comment by AKST** [comment]
Same here, if there's any grunt work that first time contributor can do to help, I'd be happy to lend a hand

**Comment by kurko** [comment]
summoning @wecc

**Comment by jvanantwerp** [comment]
For what it's worth, my company is in the same boat.  Until Ember Data fully supports the JSONAPI spec, including using related links for operations other than GET, we can't recommend Ember's full scale use.  We have data sets in the thousands so sending all the elements of a `hasMany` for simply deleting a single entry is not practical.

And we are willing to help and contribute as well.

**Comment by dbbk** [comment]
This is a fundamental dealbreaker for Ember Data, and it's coming up on 2 years since I first ran into this issue. The roadblock doesn't seem to be in implementation but just having an RFC approved. Why can't we get a decision from the core team on this?

**Comment by lvjames** [comment]
Judging by the consensus [here](https://github.com/emberjs/data/issues/2905#issuecomment-323086061) people are happy to wait indefinitely.

**Comment by dbbk** [comment]
6 months further on, still no response from the core team. If there is some protocol or path to moving this forward that I'm not aware of, someone please inform me.

**Comment by mhluska** [comment]
@dbbk  Another year later

**Comment by runspired** [comment]
@benedikt-d confirm that #332 supercedes this by solving the base set of cases. Closing.

**Pull request closed without merging**